RT #39949: enable error message for file/path not found (could also help...

rpcme opened the pull request:
... debug race condition) -- also autofix a couple ending file spaces

Commit:
RT #39949: enable error message for file/path not found (could also help debug race condition) -- also autofix a couple ending file spaces

sshaw committed:
Add .travis.yml

rpcme committed:
Merge pull request #1 from sshaw/master

Add .travis.yml

sshaw committed:
RT #73840: Minor bug in rmtree with newlines in CWD

tommylutz committed:
Fix issue 53178: blessed object treated as named arguments

tommylutz committed:
Issue 53178: blessed object treated as named arguments

theryee committed:
Bug #103512 for File-Path: Extremely minor POD fix for File::Path

rpcme committed:
Update README

Added badges.

Commit:
add md link for rendering

Commit:
Merge branch 'master' of https://github.com/rpcme/File-Path

rpcme committed:
Merge pull request #2 from tommylutz/fix-bug-53178

Fix bug 53178: object treated as named arguments

rpcme committed:
Merge pull request #3 from sshaw/RT_73840

RT #73840: Minor bug in rmtree with newlines in CWD

rpcme committed:
Merge pull request #4 from theryee/master

Bug #103512 for File-Path: Extremely minor POD fix for File::Path

Commit:
add repo location RT 95150

Commit:
Merge branch 'master' of https://github.com/rpcme/File-Path

sshaw committed:
Skip newline path tests on Windows

sshaw committed:
Update docs to compare remove_tree to `rd /s` on Win

rpcme committed:
Merge pull request #5 from sshaw/RT_73840

Skip newline path tests on Windows

theryee committed:
https://rt.cpan.org/Public/Bug/Display.html?id=72256
 #nyhackathon

rpcme committed:
Merge pull request #7 from sshaw/RT_70938

RT 70938: Update docs to compare remove_tree to `rd /s` on Win

bulk88 committed:
test UNC paths on Win32

This patch proves RT #70657 is not reproducable and is therefore rejected.

rpcme committed:
Merge pull request #6 from bulk88/master

test UNC paths on Win32

bulk88 committed:
add retval to usage examples POD

rpcme committed:
Merge pull request #9 from bulk88/master

add retval to usage examples POD

bulk88 committed:
better build product git ignoring

If a .t fails early, there wont be cleanup and a test-* folder will be
left over.

rpcme committed:
Merge pull request #10 from bulk88/master

better build product git ignoring

Commit:
fix repos path

Commit:
Merge branch 'master' of https://github.com/rpcme/File-Path

Commit:
manual merge number of tests

Commit:
Merge branch 'theryee-master'

Commit:
add contributors section from the NYC Perl Hackathon 2015

Commit:
explicit change list for upcoming version.

Commit:
add new test file, specifically for the UNIVERSAL::isa deprecation

jkeenan commented on t/Path-Class.t:
```diff
@@ -0,0 +1,49 @@
+  }
+}
+
+done_testing;
```
The use of Test::More::done_testing() will limit the backwards-compatibility of this distribution with respect to earlier versions of Perl 5.  According to the Changes file is the  Test-Simple distribution, done_testing() was introduced in Test-Simple version 0.87_01  Sun Mar 29 09:56:52 BST 2009.  So, per 'corelist -a Test::More', the earliest production version of Perl 5 that will be handle done_testing() without a problem will be 5.10.1.

I recommend counting the tests and declaring a plan.  Also, it would be a good idea to use 'perlbrew' or one of its allegedly superior replacements to locally install some earlier versions of Perl 5 and test the File-Path test suite against it.

Thank you very much.
Jim Keenan

rpcme committed:
skip test if Path::Class is not available, count tests otherwise

rpcme committed:
fix meta merge and clean up declarations

karenetheridge commented:
This PR desperately needs some rebasing.

rpcme commented:
Hi Karen,

I would love to! Just need to know what you mean by that; to me, rebasing is pulling in and merging change from the parent.  If that needs to be done, I would be happy to!  I wasn't aware there were changes in the dland repo since the hackathon.

Commit:
bump version

Commit:
Merge branch 'master' of https://github.com/rpcme/File-Path

Commit:
make the version 'testing mode', aka release_status setting

Commit:
VERSION must be eval because it isn't a number

Commit:
Administrative changes for copyrights, authors, git repos

Commit:
manually bring in changes from https://github.com/dland/File-Path/pull/4

Commit:
add appropriate SKIP test on VMS

Commit:
version bump and add deltas to the Changes file

Commit:
fix typo

jkeenan committed:
Note when 'rm_tree' was changed to 'remove_tree'.

Confirmed by inspection of commit 106ac49f1d8a800a2fcf680f91cad3c89d73eedc

For: https://rt.cpan.org/Ticket/Display.html?id=95149; ether++

rpcme committed:
Merge pull request #13 from jkeenan/95149-changes

Note when 'rm_tree' was changed to 'remove_tree'.

Commit:
more administrative changes, fix annoying camel case variables, fix symlink test to properly create test files, move File::Path to more correct directory

Commit:
further clean up the symlink test dummy file

jkeenan committed:
Provide email address for bug reports.

Provide instructions about attaching patches rather than inlining them.

rpcme committed:
Merge pull request #14 from jkeenan/rt_cpan_via_email

Provide email address for bug reports.

jkeenan committed:
Remove t/pod.t from distribution.

Update MANIFEST via 'make manifest' (hence, revised order of files).

For https://rt.cpan.org/Ticket/Display.html?id=105526

jkeenan committed:
Two distribution structure corrections.

Forgot to remove Path.pm when moved to lib/File/Path.pm.

.gitignore, .travis.yml and README.md need not go into MANIFEST.

jkeenan committed:
Two distribution structure corrections.

Forgot to remove Path.pm when moved to lib/File/Path.pm.

.gitignore, .travis.yml and README.md need not go into MANIFEST.

(cherry picked from commit 825f5415b77acf50bb149e14344e29b875606910)

jkeenan committed:
Add tests for previously unexercised conditions.

E.g., make_path and remove_tree with hash ref as final argument.

Add test for internal helper sub _slash_lc().

rpcme committed:
Merge pull request #15 from jkeenan/105526-remove-pod-test

Remove t/pod.t from distribution.

Commit:
physically remove Path.pm

Commit:
Merge branch 'correct_structure' of https://github.com/jkeenan/File-Path into jkeenan-correct_structure

Conflicts:
	MANIFEST

Commit:
Merge branch 'jkeenan-correct_structure'

rpcme committed:
Merge pull request #17 from jkeenan/5-tests

5 tests

Commit:
perl style, use English, and change bare oct to 'real' oct

Commit:
remove 'no strict refs' statement by removing strange deref -- is okay since we now support 5.8+

Commit:
clean up comment re: pre 5.6

Commit:
version bump

Commit:
add English as a dependency

Commit:
fix skip - marked 4 tests as skip when there are six

haarg committed:
remove English.pm

English.pm is only a minor readability improvement, and for people used
to perl's globals is actively confusing.  Additionally,
$OLD_PERL_VERSION was only introduced in perl 5.20, so it is useless for
how it was being used.

haarg committed:
use constants rather than lexicals for OS based checks

The OS can't change, so it's better to use constants to check it.  This
allows perl to optimize away the code for other operating systems.

haarg committed:
fix skip number when lacking Test::Output

rpcme committed:
Merge pull request #19 from haarg/no-english

remove English.pm

rpcme committed:
Merge pull request #20 from haarg/constify

use constants rather than lexicals for OS based checks

Commit:
version bump

jkeenan committed:
make_path, remove_tree:  Die on unrecognized options.

Validate each key passed in a final hashref argument to either make_path() or
remove_tree().  Assemble list of unrecognized options and die if that list has
non-zero elements, listing each bad option.

For: https://rt.cpan.org/Ticket/Display.html?id=70257

rpcme committed:
Merge pull request #21 from jkeenan/70257-check-options

make_path, remove_tree:  Die on unrecognized options.

Commit:
add count of two if Test::Output is not available skip

Commit:
remove .travis.yml and .gitignore from MANIFEST - they should not be shipped to CPAN

Commit:
release notes updated for last several dev releases and changes up until now.

Commit:
document automount and multithread limitations

Commit:
add documentation changes to the release notes

Commit:
version bump, RC

Commit:
sighs bumped the wrong version number

Commit:
Release version bump, change arg checking from croak to carp, remove tests accordingly, change skip and total test counts, edit release Changes accordingly

Commit:
do not look at swap files

jkeenan committed:
Start removing dependency on Test::Output.

Currently many test blocks in t/Path.t require CPAN module Test::Output.
While this makes for very clean code, it's unsuitable for use in a module
distributed with the Perl 5 core.  Those tests are SKIPped if Test::Output is
not present, which is always the case when running the Perl 5 core test suite
and which is often the case with CPANtesters who prize a clean environment.

But if tests are skipped, important parts of File::Path's functionality are
never tested.  Given the range of Perl versions and operating systems on which
File::Path is expected to run, this is unacceptable.

This patch starts the process of removing the dependency on Test::Output by
reverting, for 5 unit tests not run by root, to the older, admittedly less
elegant $SIG{__WARN__} approach to capturing STDERR.

jkeenan committed:
Test for unrecognized options.

In v2.10, we began to cause make_path and remove_tree to carp on unrecognized
options.  We originally wanted to have them croak, but ran into downstream
complaints, so we went with carp.  However, the tests for croaking were
commented out.

This patch replaces the commented-out tests with tests to detect warnings on
unrecognized options.

jkeenan committed:
Write helper subroutine to test warning.

This replaces use of Test::Output::stderr_is outside of locations where tests
are expected to be run as root.

jkeenan committed:
Add explicit tests for internal sub _error().

Most of the instances in which File::Path::_error() was hitherto exercised fell
into the blocks only run by root.  Hence, most of the branches and conditions
within that subroutine showed up as uncovered during coverage analysis, and it's
difficult for mere mortal users to get an idea of how the sub works.  (And
running Devel::Cover as root does not sound like a good idea.)

Add a block of tests which exercise all the nooks and crannies of
File::Path_error().

rpcme committed:
Merge pull request #22 from jkeenan/start-removing-test-output

Start removing dependency on Test::Output.

jkeenan committed:
Replace Test::Output::stdout_is() with _run_for_verbose().

This leaves Test::Output functions only in the blocks run as root.

Thanks to Zefram++ et al on perl.cpan.workers for assistance.

Commit:
Merge branch 'test-warn-unrecognized-options' of https://github.com/jkeenan/File-Path into jkeenan-test-warn-unrecognized-options

Conflicts:
	t/Path.t

Commit:
Merge branch 'jkeenan-test-warn-unrecognized-options'

Commit:
Merge branch 'demonstrate_error' of https://github.com/jkeenan/File-Path into jkeenan-demonstrate_error

Conflicts:
	t/Path.t

Commit:
Merge branch 'jkeenan-demonstrate_error'

Commit:
Merge branch 'replace_stdout_is' of https://github.com/jkeenan/File-Path into jkeenan-replace_stdout_is

Conflicts:
	t/Path.t

Commit:
Merge branch 'jkeenan-replace_stdout_is'

jkeenan committed:
Some cleanup on _run_for_verbose().

Commit:
try new test count, change some skip counts, remove test count comments that were not supposed to stay

Commit:
try another count - if this doesn't work, setup environments to cover Test::Output installed/not installed and root/non-root

Commit:
one last try

jkeenan committed:
Localize *STDOUT in _run_for_verbose().

Per suggestion by Aristotle Pagaltzis.

jkeenan committed:
Remove last instances of Test::Output::stderr_is.

jkeenan committed:
Remove last traces of Test::Output.

jkeenan committed:
Correct INSTALLDIRS setting in Makefile.PL.

Change should make latest File::Path installable on perl-5.6.2

For: https://rt.cpan.org/Ticket/Display.html?id=105977

Commit:
split out tests to remove matrix issues and move extra tests to author test only directory

rpcme committed:
Merge pull request #28 from jkeenan/105977-perl-5.6.2

Correct INSTALLDIRS setting in Makefile.PL.

Commit:
Merge branch '20150719_remove_all_test_output' of https://github.com/jkeenan/File-Path into jkeenan-20150719_remove_all_test_output

Conflicts:
	t/Path.t

Commit:
Merge branch 'jkeenan-20150719_remove_all_test_output'

Commit:
common functions to use across tests

Commit:
a more complete setup script - with teardown, and in correct author test directory

Commit:
extra tests refactored - remove symlink skip because the test bed can't be setup anyway without it :p

Commit:
use common module

Commit:
use central function module

Commit:
reflect new files and reorder a bit alphabetically

Commit:
clarify skip status reason, clean up root-only test function with import, cleanup imports

Commit:
clarify relevance of rm -r and state malformed input directory will cause error

Commit:
moved setup to better organized xt/setup_tests.pl

Commit:
update changes so far to next dev release and version bump Path.pm.

Commit:
Clean up error message for not being able to stat directory on rmtree

Commit:
RT 106039 update release version number.

Commit:
use SelectSaver (for now) and fix the multiple warning issue

haarg committed:
strict+warnings in test module

haarg committed:
5.6 compatibility in tests

Commit:
extend test matrix to 5.6.2

Commit:
add travis ci helpers

Commit:
more travis ci tweaks

rpcme committed:
Merge pull request #29 from haarg/5.6-compat

5.6 compatible testing

Commit:
run tests for all dependent modules

Commit:
bump version in pod

Commit:
Merge branch 'master' of https://github.com/rpcme/File-Path

Commit:
remove this because of the dependency - woops - keep it priv

Commit:
final prep Changes for 2.11_001

rpcme committed:
fix problem where user and group stucture sub was called multiple times, causing test failure.

rpcme committed:
Merge branch 'master' of https://github.com/rpcme/File-Path

rpcme committed:
reset passwd and group parsing in case in future we need to run again -- that way we don't trip over the frustration again

Commit:
add import that was missed after refactor

Commit:
Merge branch 'master' of https://github.com/rpcme/File-Path

rpcme committed:
further fix win32 test - since we refactored this out, it's actually creating 2 directories. Also, we need Cwd as well.

Commit:
bump version

Commit:
remove README.md (which is a symlink) and TODO which has been transferred to RT RFEs

Commit:
TODO file contents transferred to RT RFE tickets where they have better visibility and can be managed/debated better

karenetheridge commented:
I would suggest closing this pull request and opening a new one _from a separate branch_ -- if you create one from master, then whenever you change master to sync up with upstream, you're corrupting the PR.  A pull request should consist of just a few commits, with linear structure, and not contain other stuff from other authors/PRs mixed in.
